src/app/dynamic/enterprise/quotas/component.spec.ts

```diff
@@ -0,0 +1,205 @@
+    component = fixture.componentInstance;
+
+    settingService = TestBed.inject(SettingsService);
+    userService = TestBed.inject(UserService);
+
+    fixture.detectChanges();
+  });
+
+  it('should initialize', async () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should display table when quotas are loaded', async () => {
+    component.isLoading = false;
+    component.quotas = await getQuotas();
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('table'));
+
+    expect(element.nativeElement.hidden).toEqual(false);
+  });
+
+  it('should hide table when quotas are loading', async () => {
+    component.isLoading = true;
+    component.quotas = await getQuotas();
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('table'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should not display table when quotas are loaded empty', async () => {
+    component.isLoading = false;
+    component.quotas = [];
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('table'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should not display loading spinner when quotas are loading', () => {
+    component.isLoading = true;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-loading-spinner'));
+
+    expect(element).toBeTruthy();
+  });
+
+  it('should hide loading spinner when quotas are loaded', () => {
+    component.isLoading = false;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-loading-spinner > mat-spinner'));
+
+    expect(element).toBeFalsy();
+  });
+
+  it('should display correct message when loaded quotes are empty', () => {
+    component.isLoading = false;
+    component.quotas = [];
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotes-not-found'));
+
+    expect(element.nativeElement.textContent.trim()).toEqual('No Quotas Found');
+  });
+
+  it('should display paginator when received quotes are more than the user defined page size', () => {
+    const mockLength = 5;
+    component.quotas.length = mockLength - 1;
+    component.paginator.pageSize = mockLength;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-paginator'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should hide paginator when received quotes are less than the user defined page size', () => {
+    const mockLength = 5;
+    component.quotas.length = mockLength;
+    component.paginator.pageSize = mockLength + 1;
+    fixture.detectChanges();
+
+    const element = fixture.debugElement.query(By.css('#quotas-paginator'));
+
+    expect(element.nativeElement.hidden).toEqual(true);
+  });
+
+  it('should set quotas when ngOnInit is called', async () => {
```

Comment:
Instead of checking `component.quotas` its better if we check that the values are displayed in table (or similar)

Reply:
I've added unit tests for checking template too

Reply:
We can remove this test then